File: app/src/main/java/com/blues/module_main/video/VideoHolder.java
package com.blues.module_main.video;

import android.view.ViewGroup;

import com.blues.R;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;
import com.xiao.nicevideoplayer.NiceVideoPlayer;
import com.xiao.nicevideoplayer.TxVideoPlayerController;

/**
 * User : Blues
 * Date : 2019/5/17
 * Time : 10:39
 */

public class VideoHolder extends BaseViewHolder<VideoInfoEntity> {

    private NiceVideoPlayer nvPlayer;

    public VideoHolder(ViewGroup parent) {
        super(parent, R.layout.item_video_mp);

        nvPlayer = $(R.id.nvp_video);
    }

    @Override
    public void setData(VideoInfoEntity data) {
        super.setData(data);
        nvPlayer.setPlayerType(NiceVideoPlayer.TYPE_NATIVE);
        nvPlayer.setBackgroundResource(R.drawable.shape_background);
        nvPlayer.setUp(data.getVideo_url(), null);

        TxVideoPlayerController controller = new TxVideoPlayerController(getContext());
        controller.setTitle(data.getVideo_title());
        controller.setImage(R.color.color_black);
        controller.setShouldShowLength(false);
        nvPlayer.setController(controller);

        //自动播放
//        nvPlayer.post(() -> {
//            if (nvPlayer != null) {
//                nvPlayer.continueFromLastPosition(false);
//                if (nvPlayer.isIdle()) {
//                    nvPlayer.start();
//                }
//            }
//        });
    }
}

File: app/src/main/java/com/blues/gankio/v1/view/VideoHolder.kt
package com.blues.gankio.v1.view

import android.view.ViewGroup
import com.blues.gankio.v1.model.VideoInfoBean
import com.blues.R
import com.jude.easyrecyclerview.adapter.BaseViewHolder
import com.xiao.nicevideoplayer.NiceVideoPlayer
import com.xiao.nicevideoplayer.TxVideoPlayerController

/**
 * User : Blues
 * Date : 2019/5/17
 * Time : 10:39
 */
class VideoHolder(parent: ViewGroup?) :
    BaseViewHolder<VideoInfoBean>(parent, R.layout.item_video_mp) {

    private val nvPlayer: NiceVideoPlayer = itemView.findViewById(R.id.nvp_video)

    override fun setData(data: VideoInfoBean) {
        super.setData(data)
        nvPlayer.apply {
            setPlayerType(NiceVideoPlayer.TYPE_NATIVE)
            setBackgroundResource(R.drawable.shape_background)
            setUp(data.video_url, null)
            setController(TxVideoPlayerController(context).apply {
                setTitle(data.video_title)
                setImage(R.color.color_black)
                setShouldShowLength(false)
            })
        }
    }
}